incubator/beast/templates/configmap.yaml

```diff
@@ -5,33 +5,46 @@ metadata:
   BQ_WORKER_POOL_SIZE: "{{ .Values.configArgs.bqWorkerPoolSize }}"
-  GOOGLE_CREDENTIALS: "{{ .Values.configArgs.googleCredentials }}"
+  BQ_WORKER_POLL_TIMEOUT_MS: "{{ .Values.configArgs.bqWorkerPollTimeoutMs | default 5000 }}"
+  ENABLE_BQ_TABLE_PARTITIONING: "{{ .Values.configArgs.enableBqTablePartitioning }}"
+  ENABLE_BQ_ROW_INSERTID: "{{ .Values.configArgs.enableBqRowInsertid }}"
+  ENABLE_AUTO_SCHEMA_UPDATE: "{{ .Values.configArgs.enableAutoSchemaUpdate | default false }}"
   KAFKA_CONSUMER_BOOTSTRAP_SERVERS: "{{ .Values.configArgs.kafkaConsumerBootstrapServers }}"
   KAFKA_CONSUMER_GROUP_ID: "{{ include "beast.fullname" . }}"
-  KAFKA_CONSUMER_KEY_DESERIALIZER: org.apache.kafka.common.serialization.ByteArrayDeserializer
+  KAFKA_CONSUMER_KEY_DESERIALIZER: "org.apache.kafka.common.serialization.ByteArrayDeserializer"
   KAFKA_CONSUMER_MAX_POLL_RECORDS: "{{ .Values.configArgs.kafkaConsumerMaxPollRecords }}"
   KAFKA_CONSUMER_METADATA_MAX_AGE_MS: "{{ .Values.configArgs.kafkaConsumerMetadataMaxAgeMs }}"
   KAFKA_CONSUMER_SESSION_TIMEOUT_MS: "{{ .Values.configArgs.kafkaConsumerSessionTimeoutMs }}"
   KAFKA_CONSUMER_VALUE_DESERIALIZER: "org.apache.kafka.common.serialization.ByteArrayDeserializer"
   KAFKA_TOPIC: "{{ .Values.configArgs.kafkaTopic }}"
-  LOG_LEVEL: "{{ .Values.configArgs.logLevel }}"
+  KAFKA_CONSUMER_AUTO_OFFSET_RESET: "{{ .Values.configArgs.kafkaConsumerAutoOffsetReset }}"
   PROTO_COLUMN_MAPPING: '{{ .Values.configArgs.protoColumnMapping }}'
   PROTO_SCHEMA: "{{ .Values.configArgs.protoSchema }}"
+  CONSUMER_POLL_TIMEOUT_MS: "{{ .Values.configArgs.consumerPollTimeoutMs }}"
+  EXPONENTIAL_BACKOFF_INITIAL_BACKOFF_IN_MS: "{{ .Values.configArgs.exponentialBackoffInitialBackoffInMs }}"
+  EXPONENTIAL_BACKOFF_MAXIMUM_BACKOFF_IN_MS: "{{ .Values.configArgs.exponentialBackoffMaximumBackoffInMs }}"
   READ_QUEUE_CAPACITY: "{{ .Values.configArgs.readQueueCapacity }}"
   COMMIT_QUEUE_CAPACITY: "{{ .Values.configArgs.commitQueueCapacity }}"
-  STATSD_ENABLED: "true"
-  STATSD_HOST: "localhost"
-  STATSD_PORT: "8125"
-  STATSD_PREFIX: "{{ .Values.configArgs.statsdPrefix }}"
-  STENCIL_URL: "{{ .Values.configArgs.stencilURL }}"
-  SENTRY_DSN: "{{ .Values.configArgs.sentryDsn | default " " }}"
-  SENTRY_SAMPLE_RATE: {{ .Values.configArgs.sentrySampleRate | default 1.0 | quote }}
-  OFFSET_ACK_TIMEOUT: {{ .Values.configArgs.offsetAckTimeout | default 60000 | quote }}
-  BQ_WORKER_POLL_TIMEOUT_MS: {{ .Values.configArgs.bqWorkerPollTimeout | default 5000 | quote }}
-  {{ if .Values.configArgs.enableAutoSchemaUpdate }}
-  PROTO_COLUMN_MAPPING_URL: {{ .Values.configArgs.protoColumnMappingURL | default nil }}
-  UPDATE_TABLE_SCHEMA_URL: {{ .Values.configArgs.updateTableSchemaURL }}
+  LOG_LEVEL: "{{ .Values.configArgs.logLevel }}"
+  ENABLE_GCS_ERROR_SINK: "{{ .Values.configArgs.enableGcsErrorSink }}"
+  {{ if .Values.configArgs.enableGcsErrorSink }}
+  GCS_BUCKET: "{{ .Values.configArgs.gcsBucket }}"
```

Comment:
you can mark these as required if `ENABLE_GCS_ERROR_SINK` is true. 
You can set these as required like this:
`GCS_BUCKET: {{ required "A valid gcs bucket is required if gcs error sink is enabled"  .Values.configArgs.gcsBucket }}`